GliaWidgets/SecureConversations/SecureConversations.ChatWithTranscriptModel.swift

```diff
@@ -505,6 +508,7 @@ extension SecureConversations {
         }
 
         func start() {
+            environment.startSocketObservation()
```

Comment:
Where is the observation stopped? [In my PR](https://github.com/salemove/ios-sdk-widgets/pull/592/files#diff-8f58d92d9bfc067e9074401288bdb06350dd64a36e63bd024d89a2e38b116b8bR480) about system messages, I have created an event that gets called on the `deinit` of ChatViewController so that the observation is stopped at that point. If you don't want to deal with that, you can leave it and it would be added by my system messages PR, but we have to decide about it.

Reply:
I think the observation has to be stopped on upgrade to engagement, because ChatViewModel will replace TranscriptModel.